objects/src/block/mod.rs

```diff
@@ -102,6 +104,15 @@ impl Block {
         })
     }
 
+    /// Returns a note tree containing all notes created in this block.
+    pub fn build_note_tree(&self) -> Result<BlockNoteTree, MerkleError> {
+        let entries = self
+            .notes()
+            .map(|(note_index, note)| (note_index, note.id().into(), *note.metadata()));
+
+        BlockNoteTree::with_entries(entries)
+    }
```

Comment:
I think we should return just `BlockNoteTree` from here and handle error detection at construction/deserialization time. I would probably introduce `BlockError`. For now, we could check for:

- Number of batches is smaller than the allowed maximum.
- Number of notes in each batch is smaller than the allowed maximum.
- There are no duplicate notes (i.e., all notes have unique IDs).

In the future we'll add other validations (will be addressed as a part of #398).

Reply:
Thank you, I've implemented this.